### ISSUE TITLE
[BUG]: DockerCompose@0 no longer functions after removal of Docker Compose v1 from hosted agents.

### ISSUE THREAD

**Issue by trs1991** [microsoft/azure-pipelines-tasks#19711]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

DockerCompose@0

### Task version

0.231.0

### Issue Description

Now that the removal of docker compose v1 has begun rolling out as per [this GitHub issue](https://github.com/actions/runner-images/issues/9557)

The dockercompose@0 task immediately fails with '##[error]Unhandled: Docker Compose was not found. You can provide the path to docker-compose via 'dockerComposePath' due to using the legacy docker-compose v1 command that no longer exists on the Microsoft hosted agents.

The task needs to be updated to use the newer 'docker compose' v2 command.

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Ubuntu 22.04, Ubuntu 20.04, Windows Server 2019, Windows Server 2022.

### Relevant log output

```shell
2024-04-02T15:02:02.8114252Z ##[section]Starting: Build Next app container images
2024-04-02T15:02:02.8120854Z ==============================================================================
2024-04-02T15:02:02.8120983Z Task         : Docker Compose
2024-04-02T15:02:02.8121080Z Description  : Build, push or run multi-container Docker applications. Task can be used with Docker or Azure Container registry.
2024-04-02T15:02:02.8121234Z Version      : 0.231.0
2024-04-02T15:02:02.8121300Z Author       : Microsoft Corporation
2024-04-02T15:02:02.8121407Z Help         : https://aka.ms/azpipes-docker-compose-tsg
2024-04-02T15:02:02.8121504Z ==============================================================================
2024-04-02T15:02:03.9655006Z ##[error]Unhandled: Docker Compose was not found. You can provide the path to docker-compose via 'dockerComposePath' 
2024-04-02T15:02:03.9669164Z ##[error]Error: Docker Compose was not found. You can provide the path to docker-compose via 'dockerComposePath' 
    at DockerComposeConnection.setDockerComposePath (/home/vsts/work/_tasks/DockerCompose_6975e2d1-96d3-4afc-8a41-498b5d34ea19/0.231.0/dockercomposeconnection.js:166:23)
    at new DockerComposeConnection (/home/vsts/work/_tasks/DockerCompose_6975e2d1-96d3-4afc-8a41-498b5d34ea19/0.231.0/dockercomposeconnection.js:22:14)
    at success (/home/vsts/work/_tasks/DockerCompose_6975e2d1-96d3-4afc-8a41-498b5d34ea19/0.231.0/dockercompose.js:35:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2024-04-02T15:02:03.9673048Z ##[section]Finishing: Build Next app container images
```

### Full task logs with system.debug enabled

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Repro steps

```yml
steps:
- task: DockerCompose@0
  displayName: 'Build container images'
  inputs:
    containerregistrytype: 'Azure Container Registry'
    action: Build services
    projectName: '$(DOCKER-IMAGE-NAME)'
    dockerComposeFile: docker-compose-ci.yml
    qualifyImageNames: true
    additionalImageTags: '$(DOCKER-IMAGE-TAG)'
    includeLatestTag: true
```

**Comment by cgraf-spiria** [comment]
As a workaround until the task is fixed, add this to the task's inputs:
```
dockerComposePath: '/usr/libexec/docker/cli-plugins/docker-compose'
```
Also see: https://github.com/microsoft/azure-pipelines-tasks/issues/17808

**Comment by benlings** [comment]
Docker Compose v2 is also much more strict with project names (eg cannot contain upper case characters), so this bug also needs to be addressed https://github.com/microsoft/azure-pipelines-tasks/issues/16361

**Comment by fbuser1** [comment]
Wow, my entire company is blown away after this change 😡

Why didn't Microsoft issue any warnings months before the change? For example: 'This task will be deprecated, please migrate to docker-compose v2 before the 1st of April.'

Currently, I have to update about 500 microservices names from 'Company.Project.ServiceName' to 'company-project-servicename', which affects a lot of components/configs/DockerFiles entrypoints

**Comment by alpapad01** [comment]
docker-compose is deprecated. The functionality is provided by docker command itself. the command should be :

docker, with arguments compose ... rest of args

**Comment by laurencenairne-work** [comment]
@fbuser1, would [this](https://github.com/microsoft/azure-pipelines-tasks/issues/16361#issuecomment-1136364107) not solve the problem for you without needing to rename and repath everything?

**Comment by peruzzof** [comment]
My company is facing the same. Could the task be adjusted to support docker compose v2?

**Comment by alpapad01** [comment]
This works:

> dockerComposePath: '/usr/libexec/docker/cli-plugins/docker-compose'

that's the cli plugin, which the command "docker compose" would call. Apparently, it is a binary on its own.

Other options I guess, is to have a step to install the "old" docker-compose somewhere (download and install it) as shown above....

**Comment by sandywangkpmg** [comment]
dockerComposePath: '/usr/libexec/docker/cli-plugins/docker-compose' - this is v2, after switching to this path, we started seeing intermittent failure on pulling base image.
![image](https://github.com/microsoft/azure-pipelines-tasks/assets/106123825/f6c4a763-1ad3-473d-963d-e0f98651b6ab)
then I checked the image, the v1 executable is still available just not included in the PATH.
So I changed to dockerComposePath: '/usr/local/bin/docker-compose', it's working for us now.

**Comment by SwapnaVersion1** [comment]
Can we have updates on this? As Docker@0  is deprecating in July. Can we start using Docker@2 in our ADO tasks?

**Comment by ashishbhatt-afk** [comment]
> Can we have updates on this? As Docker@0 is deprecating in July. Can we start using Docker@2 in our ADO tasks?

There is no other version of the task other than 0. Am I missing something here?
![image](https://github.com/microsoft/azure-pipelines-tasks/assets/77200852/2ac08c1a-327c-42ab-adf9-73dd7d613c53)

**Comment by sandywangkpmg** [comment]
> > Can we have updates on this? As Docker@0 is deprecating in July. Can we start using Docker@2 in our ADO tasks?
> 
> There is no other version of the task other than 0. Am I missing something here? ![image](https://private-user-images.githubusercontent.com/77200852/337579294-2ac08c1a-327c-42ab-adf9-73dd7d613c53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NDg2NzMsIm5iZiI6MTcxNzc0ODM3MywicGF0aCI6Ii83NzIwMDg1Mi8zMzc1NzkyOTQtMmFjMDhjMWEtMzI3Yy00MmFiLWFkZjktNzNkZDdkNjEzYzUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA3VDA4MTkzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhNDkxY2Y4ZDhmNWFjYWViNmRkODU4YWQwMzhiMmVmMTE4Y2Y1MjU5Yzg2MWU3ZTYyNjJkZmJjNWNiYzRkYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kN0edhOA2VDijw77OBkSqXbBhuNSsbCuj8_sUmvcWAA)

There is no DockerCompose@1 or @2 version, only @0 is available. 
<img width="383" alt="image" src="https://github.com/microsoft/azure-pipelines-tasks/assets/106123825/6cb71e7c-7293-46b0-8c03-dacdbaa9db43">
I think @Jacko1394  @SwapnaVersion1 were referring to Docker task, there are @0/1/2 versions for Docker task, but it does't look like compose is a supported command.
<img width="724" alt="image" src="https://github.com/microsoft/azure-pipelines-tasks/assets/106123825/f0350f39-7a15-4af5-a066-e1bb27db1621">

**Comment by SwapnaVersion1** [comment]
What is the work-around? Our organisation pipelines would fail after July once @1 docker compose is deprecated.

**Comment by sandywangkpmg** [comment]
> > > > Can we have updates on this? As Docker@0 is deprecating in July. Can we start using Docker@2 in our ADO tasks?
> > > 
> > > 
> > > There is no other version of the task other than 0. Am I missing something here? ![image](https://private-user-images.githubusercontent.com/77200852/337579294-2ac08c1a-327c-42ab-adf9-73dd7d613c53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NDg2NzMsIm5iZiI6MTcxNzc0ODM3MywicGF0aCI6Ii83NzIwMDg1Mi8zMzc1NzkyOTQtMmFjMDhjMWEtMzI3Yy00MmFiLWFkZjktNzNkZDdkNjEzYzUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA3VDA4MTkzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhNDkxY2Y4ZDhmNWFjYWViNmRkODU4YWQwMzhiMmVmMTE4Y2Y1MjU5Yzg2MWU3ZTYyNjJkZmJjNWNiYzRkYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kN0edhOA2VDijw77OBkSqXbBhuNSsbCuj8_sUmvcWAA)
> > 
> > 
> > There is no DockerCompose@1 or @2 version, only @0 is available. <img alt="image" width="383" src="https://private-user-images.githubusercontent.com/106123825/337583967-6cb71e7c-7293-46b0-8c03-dacdbaa9db43.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NDkyNTksIm5iZiI6MTcxNzc0ODk1OSwicGF0aCI6Ii8xMDYxMjM4MjUvMzM3NTgzOTY3LTZjYjcxZTdjLTcyOTMtNDZiMC04YzAzLWRhY2RiYWE5ZGI0My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QwODI5MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01OGIxNzJlYjUwNmRjYmVlZjU4ZDA1YWI3NmE0YmE1OTYzNzFmZTNhMjVlNmZjNGUxOWNlZjVkY2JlYmRmMjRiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OK2Aj6feFe-nAb_GsYde-QwMGgA-4t2-zNy77Z6WhI8"> I think @Jacko1394 @SwapnaVersion1 were referring to Docker task, there are @0/1/2 versions for Docker task, but it does't look like compose is a supported command. <img alt="image" width="724" src="https://private-user-images.githubusercontent.com/106123825/337584920-f0350f39-7a15-4af5-a066-e1bb27db1621.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NDkyNTksIm5iZiI6MTcxNzc0ODk1OSwicGF0aCI6Ii8xMDYxMjM4MjUvMzM3NTg0OTIwLWYwMzUwZjM5LTdhMTUtNGFmNS1hMDY2LWUxYmIyN2RiMTYyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QwODI5MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZjcxM2ZkZmNkN2JhYTAzNWFmNWQzMDM1YTY0ZGJjNDZiMTU1ODFhZjJhYWZmYTE5MTNmMzEyZmMxM2U1NTliJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JTQSN5tIMtPhTqW9vySYX1uYLINaqkZzIrGoO31mFZU">
> 
> Yes that is the whole point of the original request, there is no DockerCompose@1/2, but since DockerCompose@0 is now deprecated, there is a need for it to be developed so that support for DockerCompose v2 and above can be implemented as DockerCompose v1 has been removed from hosted agents.

yep, I understood the point of this issue. I was just answering @ashishbhatt-afk's question.
And pointing out Docker@0/1/2 don't have compose command, so using Docker task is no help.

**Comment by sandywangkpmg** [comment]
> What is the work-around? Our organisation pipelines would fail after July once @1 docker compose is deprecated.

@SwapnaVersion1 You can use this workaround, it worked for us.
https://github.com/microsoft/azure-pipelines-tasks/issues/19711#issuecomment-2032731303


Bear in mind, make sure you have put depends_on when there is dependent image in your docker compose yaml file, otherwise, you would be seeing the error we were seeing due to docker compose v2 is building images in parallel. 

> dockerComposePath: '/usr/libexec/docker/cli-plugins/docker-compose' - this is v2, after switching to this path, we started seeing intermittent failure on pulling base image. ![image](https://private-user-images.githubusercontent.com/106123825/319396448-f6c4a763-1ad3-473d-963d-e0f98651b6ab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NDk2NTQsIm5iZiI6MTcxNzc0OTM1NCwicGF0aCI6Ii8xMDYxMjM4MjUvMzE5Mzk2NDQ4LWY2YzRhNzYzLTFhZDMtNDczZC05NjNkLWUwZjk4NjUxYjZhYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QwODM1NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xM2FhMWUyZjI0NGU3OTAwZjU0N2NiZTcxZmYwMmI0Yzg0OTg3YjdjMDY5ZGI2NTE1MTBjMjk1MTA5MTkzZGVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.nByCR5qYg_Wp3pVJJXUmTNUZJsus5pD8boHqhDm3TFo) then I checked the image, the v1 executable is still available just not included in the PATH. So I changed to dockerComposePath: '/usr/local/bin/docker-compose', it's working for us now.

**Comment by alec-pinson** [comment]
We started getting the following error for windows yesterday/day before:-
```
Starting: Build Docker Image
==============================================================================
Task         : Docker Compose
Description  : Build, push or run multi-container Docker applications. Task can be used with Docker or Azure Container registry.
Version      : 0.240.2
Author       : Microsoft Corporation
Help         : https://aka.ms/azpipes-docker-compose-tsg
==============================================================================
##[error]The process 'C:\Windows\system32\docker.exe' failed with exit code 15
```

fix was again the `dockerComposePath`

```
dockerComposePath: C:\ProgramData\docker-compose\docker-compose.exe
```

**Comment by sandywangkpmg** [comment]
Saw this message appeared in console output from yesterday.
We've set the dockerComposePath to use V2, to avoid surprise during the removal of v1 executable.
##[warning]The task is utilizing Docker Compose V1, but there are plans to transition to Docker Compose V2. It's necessary to investigate how to adapt your project to Docker Compose V2. For guidance on this migration, you can refer to the official Docker Compose documentation at the provided link: https://docs.docker.com/compose/migrate/

**Comment by TripleEmcoder** [comment]
Oh my, what a fine job at change management and versioning we have here 🤡

`The DockerCompose@0 task is deprecated. The task uses docker-compose v1 which no longer receives update since July 2023. To use docker compose v2, use the DockerCompose@1 task.`

That would be reasonable, if not for the fact that none of the message is true, in fact it is wrong in two separate ways:
- there is no DockerCompose@1 task published in DevOps at the moment at all and work on it according to code in this repository started only 3 days ago
- DockerCompose@0 already silently switched from `docker-compose` (v1) to `docker compose` (v2) breaking a whole lot of existing pipelines

Well, at least today it happened with a change in the patch part of the task's version number (`0.242.0` to  `0.242.8`), which is a slight improvement from last week when the new behavior hit and presented for ca. 24 h under and *unchanged version number*.

**Comment by sandywangkpmg** [comment]
> Oh my, what a fine job at change management and versioning we have here 🤡
> 
> `The DockerCompose@0 task is deprecated. The task uses docker-compose v1 which no longer receives update since July 2023. To use docker compose v2, use the DockerCompose@1 task.`
> 
> That would be reasonable, if not for the fact that none of the message is true, in fact it is wrong in two separate ways:
> 
> * there is no DockerCompose@1 task published in DevOps at the moment at all and work on it according to code in this repository started only a week ago
> * DockerCompose@0 already silently switched from `docker-compose` (v1) to `docker compose` (v2) breaking a whole lot of existing pipelines
> 
> Well, at least today it happened with a small increment in the task's version number (`0.242.0` to `0.242.8`), which is a slight improvement over two weeks ago when the new behavior hit without any warning under an _unchanged version number_ (`0.240.2`) and stayed there for ca. 24 h until it was rolled back as silently as it was introduced.

Yeah, we felt the pain too. Too many main stream changes all rushed in around the same time. V1 executable will be gone from MS hosted agent on 9th July, now DockerCompose@0 will be removed (below message is from the task output)

![image](https://github.com/microsoft/azure-pipelines-tasks/assets/106123825/ad66165a-07fa-40b2-9684-5f0b622b3191)

but yet, DockerCompose@1 isn't made official yet (there was a commit to master branch 3 days ago https://github.com/microsoft/azure-pipelines-tasks/commit/d8e6a2ea34df6114e580b8b7f14413489551f59c) 
![image](https://github.com/microsoft/azure-pipelines-tasks/assets/106123825/46112a03-ecf6-4137-b6f7-283fb9af8468)

**Comment by sandywangkpmg** [comment]
https://github.com/microsoft/azure-pipelines-tasks/pull/20085 This is the ticket to add DockerComposeV1, I left a message there.

**Comment by sandywangkpmg** [comment]
ok, turned out DockerCompose@1 task was released but the commit 3 days ago didn't include docs change.
I just tested using DockerCompose@1 in my pipeline and it worked. No more warning message.
And the task version number is **1.242.1**
<img width="626" alt="image" src="https://github.com/microsoft/azure-pipelines-tasks/assets/106123825/85a82d62-1750-4994-85eb-8ec8d042a9f8">

**Comment by sandywangkpmg** [comment]
ok, just learnt that docker compose v1 executable will be removed on 29th July, giving people to migration to DockerCompose@1 before @0 gets removed on 24th July, I suppose.

![image](https://github.com/microsoft/azure-pipelines-tasks/assets/106123825/fdc9ae09-cf39-434e-953c-804cf4ab7086)

**Comment by ivanduplenskikh** [comment]
@sandywangkpmg, @TripleEmcoder, I would like to clarify what is happening here.
Initially, we planned to migrate **DockerComposeV0** task to the **docker compose CLI v2** but there were problems with project names due to [constraints for a project name](https://docs.docker.com/compose/project-name/#set-a-project-name).
After that we released the **DockerComposeV1** task that uses the **docker compose CLI v2**, but the **DockerComposeV0** task has been rolled back to the **docker-compose CLI v1**.
I agree that was a little messed up and we had a small amount of time to inform users about the new major version of the task. Now we have the mention only in [the release notes](https://learn.microsoft.com/en-us/azure/devops/release-notes/2024/pipelines/sprint-240-update#dockercompose0-uses-docker-compose-v2-in-v1-compatibility-mode) but I hope that we see an updated documentation about **DockerComposeV1** soon.
Thanks for understanding.

Related to the tasks changes:
As I mentioned above, the **DockerComposeV0** task is using compose v1 (if compose v1 is not found the task switches to compose v2) and the **DockerComposeV1** task is using compose v2.
If you want to use compose v1 in the **DockerCompose** after it will be removed from the images you can refer to [the instructions](https://learn.microsoft.com/en-us/azure/devops/release-notes/2024/pipelines/sprint-240-update#use-docker-compose-v1-on-windows) that helps you to use compose v1 with the **DockerComposeV0** task.

**Comment by merovingio96** [comment]
Hi everyone! 

In my department we have experienced an issue related with the DockerCompose@1 task. The last week we changed pipeline tasks related with te build and push to ACR images, using the DockerCompose@1.

We changed our services docker-compose.yml in order to omit the "version" statement" and begin directly with the service statements.

But we encountered this message from pipeline: 
`##[error]time="2024-07-24T13:40:37Z" level=warning msg="/home/vsts/agents/3.241.0/.docker-compose.1721828409324.yml: `version` is obsolete"`

And the docker-compose.yml seems like this:
![image](https://github.com/user-attachments/assets/ce79026e-cdb6-42e1-b99b-30171d459c9c)

What are we missing up?

**Comment by trs1991** [comment]
> Hi everyone!
> 
> In my department we have experienced an issue related with the DockerCompose@1 task. The last week we changed pipeline tasks related with te build and push to ACR images, using the DockerCompose@1.
> 
> We changed our services docker-compose.yml in order to omit the "version" statement" and begin directly with the service statements.
> 
> But we encountered this message from pipeline: `##[error]time="2024-07-24T13:40:37Z" level=warning msg="/home/vsts/agents/3.241.0/.docker-compose.1721828409324.yml: `version` is obsolete"`
> 
> And the docker-compose.yml seems like this: ![image](https://private-user-images.githubusercontent.com/133207645/351743865-ce79026e-cdb6-42e1-b99b-30171d459c9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MzAyODIsIm5iZiI6MTcyMTgyOTk4MiwicGF0aCI6Ii8xMzMyMDc2NDUvMzUxNzQzODY1LWNlNzkwMjZlLWNkYjYtNDJlMS1iOTliLTMwMTcxZDQ1OWM5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNFQxNDA2MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MmYzMDI1MDk4ZWM4ZWM2NmM1MTM0ZWVmZDYxNGNlMjdmMjRiMWFjMDQxZmI3ZWE3OTA5ZTU4OTY2ZmI0ZDQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ikKCvLf5XLIcLt80sc0WZuSsgrbxXZHCo86rufLCfME)
> 
> What are we missing up?

You're not missing anything, the DockerCompose tasks add in their own docker compose files by default as extra compose files, the task owner would need to update these to also omit the version statements.

**Comment by merovingio96** [comment]
> > Hi everyone!
> > In my department we have experienced an issue related with the DockerCompose@1 task. The last week we changed pipeline tasks related with te build and push to ACR images, using the DockerCompose@1.
> > We changed our services docker-compose.yml in order to omit the "version" statement" and begin directly with the service statements.
> > But we encountered this message from pipeline: `##[error]time="2024-07-24T13:40:37Z" level=warning msg="/home/vsts/agents/3.241.0/.docker-compose.1721828409324.yml: `version` is obsolete"`
> > And the docker-compose.yml seems like this: ![image](https://private-user-images.githubusercontent.com/133207645/351743865-ce79026e-cdb6-42e1-b99b-30171d459c9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MzAyODIsIm5iZiI6MTcyMTgyOTk4MiwicGF0aCI6Ii8xMzMyMDc2NDUvMzUxNzQzODY1LWNlNzkwMjZlLWNkYjYtNDJlMS1iOTliLTMwMTcxZDQ1OWM5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNFQxNDA2MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MmYzMDI1MDk4ZWM4ZWM2NmM1MTM0ZWVmZDYxNGNlMjdmMjRiMWFjMDQxZmI3ZWE3OTA5ZTU4OTY2ZmI0ZDQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ikKCvLf5XLIcLt80sc0WZuSsgrbxXZHCo86rufLCfME)
> > What are we missing up?
> 
> You're not missing anything, the DockerCompose tasks add in their own docker compose files by default as extra compose files, the task owner would need to update these to also omit the version statements.

Hi @trs1991! We are using the classic editor (please take a look the image in edit of my last message). How we can achieve omit or prevent DockerCompose task add its own docker compose files?

Thanks in advance
